Source/WebCore/page/playstation/ResourceUsageThreadPlayStation.cpp

```diff
@@ -55,17 +57,31 @@ void ResourceUsageThread::platformCollectCPUData(JSC::VM*, ResourceUsageData& da
 
 void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageData& data)
 {
-    // FIXME: Calculate total dirty memory size for ResourceUsageOverlay
-    data.totalDirtySize = 0;
+    auto& categories = data.categories;
+
+    data.totalDirtySize = bmalloc::api::memoryFootprint();
 
     size_t currentGCHeapCapacity = vm->heap.blockBytesAllocated();
     size_t currentGCOwnedExtra = vm->heap.extraMemorySize();
     size_t currentGCOwnedExternal = vm->heap.externalMemorySize();
     RELEASE_ASSERT(currentGCOwnedExternal <= currentGCOwnedExtra);
 
-    data.categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
-    data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
-    data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+    categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
+    categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
+    categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+
+    showmap::Result<256> result;
+    result.collect();
+    if (auto* entry = result.entry("SceNKFastMalloc"))
+        categories[MemoryCategory::bmalloc].dirtySize = entry->effectiveRss();
+
+    // TODO: collect dirty size of "MemoryCategory::Images" and "MemoryCategory::Layers"
+    if (data.totalDirtySize > categories[MemoryCategory::bmalloc].dirtySize)
+        categories[MemoryCategory::Other].dirtySize = data.totalDirtySize - categories[MemoryCategory::bmalloc].dirtySize;
+
+    auto& dirtySize = categories[isFastMallocEnabled() ? MemoryCategory::bmalloc : MemoryCategory::Other].dirtySize;
```

Comment:
Relationship between `categories[MemoryCategory::bmalloc]` and `categories[MemoryCategory::Other]` is a bit unclear.

Can it be more clear if their shape is like this?
```
if (isFastMallocEnabled()) {
   ...
} else {
   ...
}
```

Reply:
Then the `auto&` cannot be used, so I made a similar code based on cocoa port.
https://github.com/WebKit/WebKit/blob/main/Source/WebCore/page/cocoa/ResourceUsageThreadCocoa.mm#L262